src/secrets-kubernetes/src/lib.rs

```diff
@@ -33,15 +56,69 @@ impl KubernetesSecretsController {
             },
         };
         let client = Client::try_from(kubeconfig)?;
+        let secret_api: Api<Secret> = Api::default_namespaced(client);
 
-        Ok(KubernetesSecretsController {
-            _kube_client: client,
-        })
+        let name: String = format!("{}", HARDCODED_NAME);
+
+        match secret_api.get_opt(&*name).await? {
+            Some(_) => {
+                // nothing to do, secret already exists
+            }
+            None => {
```

Comment:
this makes me a bit nervous. The patch below will override any existing secret. So if someone concurrently creates a new secret and populates it with data, this code could in theory discard that data.